source-test/File.ts

```diff
@@ -1,661 +1,1617 @@
       done();
     });
+  });
 
-    it('should copy all attributes over with Stream', done => {
-      const options = {
-        cwd: '/',
-        base: '/test/',
-        path: '/test/test.coffee',
-        contents: fakeStream
-      };
-      const file = new File(options);
-      const file2 = file.clone();
+  describe('isSymbolic()', function() {
+    const fakeStat: fs.Stats = <any> {
+      isSymbolicLink: () => {
+        return true;
+      },
+    };
+
+    it('returns false when the contents are a Buffer', function(done) {
+      const val = new Buffer('test');
+      const file = new Vinyl({ contents: val, stat: fakeStat });
+      expect(file.isSymbolic()).toEqual(false);
+      done();
+    });
+
+    it('returns false when the contents are a Stream', function(done) {
+      const val = from([]);
+      const file = new Vinyl({ contents: val, stat: fakeStat });
+      expect(file.isSymbolic()).toEqual(false);
+      done();
+    });
+
+    it('returns true when the contents are null & stat.isSymbolicLink is true', function(done) {
+      const file = new Vinyl({ contents: null, stat: fakeStat });
+      expect(file.isSymbolic()).toEqual(true);
+      done();
+    });
+
+    it('returns false when stat exists but does not contain an isSymbolicLink method', function(done) {
+      const file = new Vinyl({ contents: null, stat: <fs.Stats> {} });
+      expect(file.isSymbolic()).toEqual(false);
+      done();
+    });
 
-      file2.should.not.equal(file, 'refs should be different');
-      file2.cwd.should.equal(file.cwd);
-      file2.basename.should.equal(file.basename);
-      file2.path.should.equal(file.path);
-      file2.contents.should.equal(file.contents, 'stream ref should be the same');
+    it('returns false when stat does not exist', function(done) {
+      const file = new Vinyl({ contents: null });
+      expect(file.isSymbolic()).toEqual(false);
       done();
     });
+  });
+
+  describe('clone()', function() {
 
-    it('should copy all attributes over with null', done => {
+    it('copies all attributes over with Buffer contents', function(done) {
       const options = {
         cwd: '/',
         base: '/test/',
         path: '/test/test.coffee',
-        contents: fakeStream
+        contents: new Buffer('test'),
       };
-      const file = new File(options);
+      const file = new Vinyl(options);
       const file2 = file.clone();
 
-      file2.should.not.equal(file, 'refs should be different');
-      file2.cwd.should.equal(file.cwd);
-      file2.basename.should.equal(file.basename);
-      file2.path.should.equal(file.path);
-      should.not.exist(file2.contents);
+      expect(file2).toNotBe(file);
+      expect(file2.cwd).toEqual(file.cwd);
+      expect(file2.base).toEqual(file.base);
+      expect(file2.path).toEqual(file.path);
+      expect(file2.contents).toNotBe(file.contents);
+      expect((<Buffer> file2.contents).toString('utf8')).toEqual((<Buffer> file.contents).toString('utf8'));
```

Comment:
Which style do you prefer ?

This one is the closer to the original tests:
```typescript
expect(file2.contents).toBeA(Buffer);
expect(file.contents).toBeA(Buffer);
expect((file2.contents as Buffer).toString('utf8')).toEqual((file.contents as Buffer).toString('utf8'));
```
Or this one that Typescript understands (so no need to manually add type assertion):
```typescript
if (!(file2.contents instanceof Buffer)) {
  throw new TypeError("Expected `file2.contents` to be a `Buffer`");
}
if (!(file.contents instanceof Buffer)) {
  throw new TypeError("Expected `file.contents` to be a `Buffer`");
}
expect(file2.contents.toString('utf8')).toEqual(file.contents.toString('utf8'));
```

In fact, I'm fearing that it is starting to deviate a bit too much from the original tests: we are not really running the tests, just compiling them. The fact that the cast to `Buffer` is allowed is already good enough (for example, it wouldn't let you cast to `Array`) to check the definitions.